### PR TITLE
make datastream processes and merge reader and collector

### PR DESCRIPTION
**Pull request by Pkoiralap**
PR has basically two things:

1. The reader and the collector modules are merged
2. The datastreams act as independent processes

### CODE REVIEW

File: tests/subsystems/test_reader.py [hunk 1]
```diff
@@ -75,11 +75,11 @@ def test_start_and_shutting_down(reader):
 
     # Start 
     reader.start()
-
+    time.sleep(0.5)
     # Stop
     reader.shutdown()
     reader.join()
-    assert reader.get_shutdown() == True
+    assert reader.get_running() == True
```
Comment (Pkoiralap) [inline tests/subsystems/test_reader.py:82]:
```suggestion
    assert reader.get_running() == False
```

File: tests/unit/test_process.py [hunk 2]
```diff
@@ -100,7 +100,7 @@ def test_process_shutdown():
     p1 = cp.Process(name='test', inputs=None)
 
     # Print the original memory value
-    assert p1.get_shutdown() == False
+    assert p1.get_running() == False
```
Comment (Pkoiralap) [inline tests/unit/test_process.py:103]:
```suggestion
    assert p1.get_running() == True
```

File: tests/unit/test_process.py [hunk 3]
```diff
@@ -111,8 +111,8 @@ def test_process_shutdown():
     # Wait until the process stops
     p1.join()
     
-    # assert p1.get_shutdown() == True
-    assert p1.get_shutdown() == True
+    # assert p1.get_running() == True
+    assert p1.get_running() == True
```
Comment (Pkoiralap) [inline tests/unit/test_process.py:115]:
```suggestion
    assert p1.get_running() == False
```

**Comment by edavalosanaya** [comment]
When our implementation used `SingleRunner` and `GroupRunner`, the group resembled the group of data streams for a single runner. The `user_data_streams` was organized like this: `user_data_streams: Dict[str, Dict[str, DataStream]]`, where it was like this Dict[runner.name, Dict[datastream.name, datastream]] , now that we don't use the runner classes, this distinguishing isn't necessary. Overall your changes are excellent! I would just like to expand it to correct this artifact that I didn't remove from a previous implementation.

Now that the Reader and Collector are merged, you can just keep the data streams perhaps in a simple dictionary Dict[str, DataStream] and rename user_data_streams to just dss throughout the Reader implementation?

We can make this changes in this PR or in another one, up to you. Overall, looks good.

**Review by edavalosanaya — CHANGES_REQUESTED** [review]
Good quality of life improvement changes. Maybe something good to do is to mark all the tests that aren't passing (haven't gotten to them yet because of refactoring), have become obsolete, or are just not in the scope of this RP by using pytest xfail or commenting them out. 

Because we are doing major refactoring, breaking changes are expected. But we still want the new version of ChimeraPy to pass all the tests (that we have updated to match the new implementation) and having a stable build. All following pushes should then also pass this tests and new ones.

So my only request is remove/mark failing test and keep the ones that you used during development. Want to keep CI as we refactor.

**Review by edavalosanaya — APPROVED** [review]

**Comment by Pkoiralap** [comment]
Regarding the tests, they are failing because of failing data pull. Please check the error log [here](https://github.com/oele-isis-vanderbilt/ChimeraPy/runs/6973994018?check_suite_focus=true). I think we can turn the tests off for the experimental branch altogether and only have it in the main branch. That way we can focus on building rather than testing for now.

Please stay tuned for the datastream changes in the reader. Will push that in a while

**Pull request merged**